csv/parse.ts

```diff
@@ -333,26 +347,137 @@ export function parse(input: string): string[][];
 /**
  * Parses CSV string into an array of objects or an array of arrays of strings.
  *
- * If `column` or `skipFirstRow` option is provided, it returns an array of
+ * If `columns` or `skipFirstRow` option is provided, it returns an array of
  * objects, otherwise it returns an array of arrays of string.
  *
- * @example Usage
+ * @example skipFirstRow: false
  * ```ts
  * import { parse } from "@std/csv/parse";
- * import { assertEquals } from "@std/assert";
+ * import { assertEquals } from "@std/assert/equals";
+ * import { assertType, IsExact } from "@std/testing/types"
  *
  * const string = "a,b,c\nd,e,f";
+ * const result = parse(string, { skipFirstRow: false });
+ *
+ * assertEquals(result, [["a", "b", "c"], ["d", "e", "f"]]);
+ * assertType<IsExact<typeof result, string[][]>>(true);
+ * ```
+ *
+ * @example skipFirstRow: true
+ * ```ts
+ * import { parse } from "@std/csv/parse";
+ * import { assertEquals } from "@std/assert/equals";
+ * import { assertType, IsExact } from "@std/testing/types"
+ *
+ * const string = "a,b,c\nd,e,f";
+ * const result = parse(string, { skipFirstRow: true });
+ *
+ * assertEquals(result, [{ a: "d", b: "e", c: "f" }]);
+ * assertType<IsExact<typeof result, Record<string, string | undefined>[]>>(true);
```

Comment:
I added `assertType` here because I think this is a very good document for users given that the return type varies depending on the option we pass in.

While I was writing these assertions, though, I was unsure in what cases the value in the return value becomes `undefined`. Does anyone know why we have `Record<string, string | undefined>` instead of `Record<string, string>` here?

Reply:
I tried the following three test cases and these pass. I couldn't find a case where the parsed value becomes `undefined`

```ts
    await t.step({
      name: "BlankField2",
      fn() {
        const input = "a,b,c\nd,,f";
        assertEquals(
          parse(input, { skipFirstRow: true }),
          [{ a: "d", b: "", c: "f" }],
        );
      },
    });

    await t.step({
      name: "Exessive fields with skipFirstRow: true",
      fn() {
        const input = "a,b,c\nd,,f,g";
        assertThrows(
          () => parse(input, { skipFirstRow: true }),
          Error,
          "Error number of fields line: 1\nNumber of fields found: 3\nExpected number of fields: 4",
        );
      },
    });

    await t.step({
      name: "Insufficient fields with skipFirstRow: true",
      fn() {
        const input = "a,b,c\nd,e";
        assertThrows(
          () => parse(input, { skipFirstRow: true }),
          Error,
          "Error number of fields line: 1\nNumber of fields found: 3\nExpected number of fields: 2",
        );
      },
    });
```

Reply:
Added (or fixed the existing) tests on what happens if the number of fields in records doesn't match that of the header. This fix contains the change in how we display the line number is error messages (which was 0-based but now changed to 1-based).

a91806b